core/src/main/java/jenkins/mvn/DefaultGlobalSettingsProvider.java

```diff
@@ -20,7 +22,7 @@ public DefaultGlobalSettingsProvider() {
     }
 
     @Override
-    public FilePath supplySettings(AbstractBuild<?, ?> project, TaskListener listener) {
+    public FilePath supplySettings(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull TaskListener listener) {
```

Comment:
@KostyaSha This is a method with `@Override`. So it does not break the binary compatibility